src/main.cpp

```
@@ -1320,57 +1421,60 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
     if (!CheckFinalTx(tx, STANDARD_LOCKTIME_VERIFY_FLAGS))
         return state.DoS(0, false, REJECT_NONSTANDARD, "non-final");
 
-    // is it already in the memory pool?
+
     uint256 hash = tx.GetHash();
-    if (pool.exists(hash))
-    {
-        LogPrint("mempool", "Dropping txid %s : already in mempool\n", hash.ToString());
-        return false;
-    }
 
-    // perform some check related to sidechains state, e.g. creation of an existing scid, fw to
-    // a not existing one and so on
-    if (!scMgr.IsTxApplicableToState(tx) )
+    // Check if tx is already in mempool or if there are conflicts with in-memory transactions
     {
-        return false;
-    }
-    LogPrint("sc", "%s():%d - tx [%s] is applicable\n", __func__, __LINE__, hash.ToString());
+        LOCK(pool.cs);
 
-    // Check for conflicts with in-memory transactions
-    {
-        LOCK(pool.cs); // protect pool.mapNextTx
-        for (unsigned int i = 0; i < tx.vin.size(); i++)
-        {
-            COutPoint outpoint = tx.vin[i].prevout;
-            if (pool.mapNextTx.count(outpoint))
-            {
+        if (pool.mapTx.count(hash) != 0) {
+            LogPrint("mempool", "Dropping txid %s : already in mempool\n", hash.ToString());
+            return false;
+        }
+
+        for (const CTxIn & vin : tx.GetVin()) {
+            if (pool.mapNextTx.count(vin.prevout)) {
                 // Disable replacement feature for now
+                LogPrint("mempool", "%s():%d - Dropping txid %s : it double spends input of tx[%s] that is in mempool\n",
+                    __func__, __LINE__, hash.ToString(), vin.prevout.hash.ToString());
+                return state.DoS(0,
```

Returning an error will set state.mode to "invalid" causing the caller to communicate back to the sending peer that the tx was rejected. The original behavior here was to just silently ignore this tx. Moreover the same "ignoring" behavior MUST be followed when receiving missing inputs txs because these txs can depend on other txs not yet relayed by other peers and for this reason they are kept in the global orphans list and should not be communicated as rejected.
Just return false here

---

Done